api/controllers/task.go

```diff
@@ -69,6 +82,46 @@ func (t *TaskController) GetTask(c *fireball.Context) (fireball.Response, error)
 	return fireball.NewJSONResponse(200, model)
 }
 
+func (t *TaskController) GetTaskLogs(c *fireball.Context) (fireball.Response, error) {
+	id := c.PathVariables["id"]
+	query := c.Request.URL.Query()
+
+	var tail int
+	if v := query.Get("tail"); v != "" {
+		t, err := strconv.Atoi(v)
+		if err != nil {
+			return nil, errors.Newf(errors.InvalidRequest, "Tail must be an integer")
+		}
+
+		tail = t
+	}
+
+	parseTime := func(v string) (time.Time, error) {
+		if v == "" {
+			return time.Time{}, nil
+		}
+
+		return time.Parse(TIME_LAYOUT, v)
+	}
+
+	start, err := parseTime(query.Get("start"))
+	if err != nil {
+		return nil, errors.Newf(errors.InvalidRequest, "Invalid time: start must be in format YYYY-MM-DD HH:MM")
```

Comment:
Why are we doing request verification in the `controller` layer?

Reply:
we do all request verification in the controller layer

Reply:
I'm not catching were we delineate request verification versus something else. For example, what's the difference between request verification here and this? https://github.com/quintilesims/layer0/blob/api-refactor/api/provider/aws/environment_create.go#L38

Reply:
That example is an outlier due to the nature of that particular parameter.
Regardless of whatever verification is done in the `controller` layer for the environment's os, we would still need to throw an error if we couldn't match the specified os to a user_data template in the `provider` layer.

Reply:
OK, I see what you're saying re: where the error is thrown, makes sense. 👍